src/plots/cartesian/set_convert.js

```diff
@@ -152,10 +152,8 @@ module.exports = function setConvert(ax, fullLayout) {
         if(ax._categoriesMap) {
             index = ax._categoriesMap[v];
             if(index !== undefined) return index;
-        } else {
-            index = ax._categories.indexOf(v);
-            if(index !== -1) return index;
```

Comment:
Awesome, I'm glad it was as easy as that :)

Apologies for the iterations, but one more related request since this is a perf PR of a couple of very hot code paths: I don't think that either `getCategoryIndex` or `setCategoryIndex` needs the fallbacks for missing `_categoriesMap` - the previous version, using only `_categories`, didn't have these fallbacks, and every path that creates `_categories` also creates `_categoriesMap`, right?

(also 📚  your [comment above this](https://github.com/plotly/plotly.js/pull/1544/commits/d2be6c94f3291c42666bf5c3f086968f42010ba2#diff-bddbbac00596941548be19d6840c720aL151) is out of date now)